src/vs/workbench/contrib/scm/common/quickDiffService.ts

```diff
@@ -60,8 +60,9 @@ export class QuickDiffService extends Disposable implements IQuickDiffService {
 
 	async getQuickDiffs(uri: URI, language: string = '', isSynchronized: boolean = false): Promise<QuickDiff[]> {
 		const sorted = Array.from(this.quickDiffProviders).sort(createProviderComparer(uri));
+		const withMatchingRootUri = sorted.filter(provider => !provider.rootUri || isEqualOrParent(uri, provider.rootUri));
```

Comment:
Instead of (1) sorting and (2) filtering, shouldn't we (1) filter and (2) sort?

```suggestion
		const providers = Array.from(this.quickDiffProviders)
			.filter(provider => !provider.rootUri || isEqualOrParent(uri, provider.rootUri))
			.sort(createProviderComparer(uri));
```